tests/specs/nodeRewardsProgram.spec.js

```diff
@@ -0,0 +1,373 @@
+            expect(transaction.otherTrans.recipient).toBe(enrollmentAddress);
+            expect(transaction.otherTrans.amount).toBe(0);
+            expect(transaction.otherTrans.mosaics).toBe(null);
+            expect(transaction.otherTrans.message.payload).toBe(expectedMessage);
+            done();
+        });
+    });
+
+    describe('checkEnrollmentAddress()', () => {
+        it('should return true when API returns "true" string', async (done) => {
+            // Arrange:
+            mockFetch('true');
+            const expectedResult = true;
+
+            // Act:
+            const result = await NodeRewardsProgram.checkEnrollmentAddress('');
+
+            // Assert:
+            expect(result).toBe(expectedResult);
+            done();
+        });
+
+        it('should return false when API returns "false" string', async (done) => {
+            // Arrange:
+            mockFetch('false');
+            const expectedResult = false;
+
+            // Act:
+            const result = await NodeRewardsProgram.checkEnrollmentAddress('');
+
+            // Assert:
+            expect(result).toBe(expectedResult);
+            done();
+        });
+
+        it('should throw an error when request failed', async (done) => {
+            // Arrange:
+            mockFetch(null, false);
+            const expectedError = Error('failed_to_validate_enroll_address');
+
+            // Act:
+            try {
+                await NodeRewardsProgram.checkEnrollmentAddress('');
+            }
+            catch(e) {
+                // Assert:
+                expect(e.message).toBe(expectedError.message);
+                done();
+            }
+        });
+    });
+
+    describe('checkEnrollmentStatus()', () => {
+        it('should return true when there is successfull enrollment for given public key and recipient address is a valid enrollment address', async (done) => {
+            // Arrange:
+            spyOn(NodeRewardsProgram, 'checkEnrollmentAddress').and.returnValue(Promise.resolve(true));
+            mockFetch([{
+                recipientAddress: 'NCHESTYVD2P6P646AMY7WSNG73PCPZDUQNSD6JAK',
+                transactionHeight: 1,
+                transactionHash: 'bba40188dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891aua',
+                host: 'http://108.61.168.86:7890'
+            }]);
+            
+            const publicKey = '00a30788dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891a4d';
+            const expectedEnrollmentStatus = true;
+
+            // Act:
+            const enrollmentStatus = await NodeRewardsProgram.checkEnrollmentStatus(publicKey);
+
+            // Assert:
+            expect(enrollmentStatus).toBe(expectedEnrollmentStatus);
+            done();
+        });
+
+        it('should return false when there is successfull enrollment for given public key and recipient address is not a valid enrollment address', async (done) => {
```

Comment:
can refactor first two tests as only difference is true/false

Reply:
Refactored.